modules/tinymce/src/core/main/ts/mode/Disabled.ts

```diff
@@ -0,0 +1,144 @@
+import { Arr, Optional, Strings } from '@ephox/katamari';
+import { Attribute, Compare, SelectorFilter, SelectorFind, SugarElement } from '@ephox/sugar';
+
+import Editor from '../api/Editor';
+import VK from '../api/util/VK';
+import * as ModeUtils from '../util/ModeUtils';
+
+const isDisabled = (editor: Editor): boolean => editor.disabled;
```

Comment:
Do we really need to expose that disabled state on the editor instance? Setting that manually will not work anyway since it doesn't dispatch events etc. Can it be hidden by using the option since that seems to be the mutating state being used here.

```suggestion
const isDisabled = (editor: Editor): boolean => Options.isDisabled(editor);
```

Reply:
I can see why this is here - it's a convenient way for any internal component to get the current state of the value - but I think Spocke is right, we can't confuse integrators by offering a potential second way to "set" the value. We need to rely on `editor.options.get` for internal use.

- [ ] remove editor property

Reply:
@TheSpyder @spocke  Requesting re-review, one problem I’m facing is the `DisabledStateChangeEvent` is fired before the option is changed, so when getting the value of the option with `isDisabled: () => editor.options.get('disabled')` returns false. And that is not returning the correct result when I use `isDisabled` check in things like `ObjectSelected` and in the UI backstage, because updating the option only done after the listeners have processed all the handlers. I've wrapped the dispatching of the event in a `promise.resolve().then` so that it's queued in the next queue. Let me know if it works.

Reply:
Or maybe a `postProcess` callback that executes after the value differs from the original value? 
https://github.com/tinymce/tinymce/compare/feature/TINY-11488...feature/TINY-114888_spike?expand=1

Reply:
call it `onChange` instead of `postProcess`, because it won't run unless the set option is different

log as a ticket for 7.7 or 8.0